chainer/utils/conv_nd.py

```diff
@@ -0,0 +1,126 @@
+import itertools
+import numpy
+import six
+
+from chainer import cuda
+from chainer.utils.conv import get_conv_outsize
+from chainer.utils import conv_nd_kernel
+
+
+def as_tuple(x, n):
+    if hasattr(x, '__getitem__'):
+        assert len(x) == n
+        return tuple(x)
+    return (x,) * n
+
+
+def im2col_nd_cpu(img, ksize, stride, pad, pval=0, cover_all=False):
```

Comment:
I suppose we can if we think such that `im2col_nd_cpu` is Chainer's internal utility and Chainer users do not use it directly, it is always called from some Chainer functions and perhaps links, the caller should have responsibility to check the length of them regardless of `im2col_nd_cpu` calling, so it may pass the check.

But I also think that checking it in `im2col_nd_cpu` does not cause any significant penalty. I will add the check here.